### PR TITLE
Remove support of neutron-lbaas from lb_l7policy

### PR DESCRIPTION
**Pull request by nikParasyr**
Part of: https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1638

### CODE REVIEW

File: openstack/lb_v2_octavia.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/l7policies"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/listeners"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/loadbalancers"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/monitors"
+	"github.com/gophercloud/gophercloud/openstack/loadbalancer/v2/pools"
+)
+
+func resourceLBV2ListenerRefreshFuncOctavia(lbClient *gophercloud.ServiceClient, lbID string, listener *listeners.Listener) resource.StateRefreshFunc {
+	if listener.ProvisioningStatus != "" {
```
Comment (kayrus) [inline openstack/lb_v2_octavia.go:20]: This is my code I wrote 5 years ago :) Not sure if this is the right place to ask this, but with my current experience I'd check for empty string and return `resourceLBV2LoadBalancerStatusRefreshFuncOctavia` right away.

File: openstack/lb_v2_octavia.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+		if len(listener.Loadbalancers) > 0 {
+			return listener.Loadbalancers[0].ID, nil
+		}
+	}
+
+	return "", fmt.Errorf("Unable to determine loadbalancer ID from pool %s", pool.ID)
+}
+
+func resourceLBV2PoolRefreshFuncOctavia(lbClient *gophercloud.ServiceClient, lbID string, pool *pools.Pool) resource.StateRefreshFunc {
+	if pool.ProvisioningStatus != "" {
```
Comment (kayrus) [inline openstack/lb_v2_octavia.go:95]: here as well

File: openstack/lb_v2_octavia.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+		}
+
+		return fmt.Errorf("Error waiting for l7policy %s to become %s: %s", l7policy.ID, target, err)
+	}
+
+	return nil
+}
+
+func resourceLBV2L7PolicyRefreshFuncOctavia(lbClient *gophercloud.ServiceClient, lbID string, l7policy *l7policies.L7Policy) resource.StateRefreshFunc {
+	if l7policy.ProvisioningStatus != "" {
```
Comment (kayrus) [inline openstack/lb_v2_octavia.go:285]: and here too

**Review by kayrus — COMMENTED** [review]
Thanks for the PR. See my comments

File: openstack/lb_v2_octavia.go (hunk 1 above)
Comment (nikParasyr) [reply openstack/lb_v2_octavia.go:20]: @kayrus can you elaborate a bit more?

> I'd check for empty string and return resourceLBV2LoadBalancerStatusRefreshFuncOctavia right away.

Return resourceLBV2LoadBalancerStatusRefreshFuncOctavia when == "" or != "" ? And what do we return in the other case? 
Cause after looking a bit the code i think we can always return resourceLBV2LoadBalancerStatusRefreshFuncOctavia no matter the check, but maybe im misunderstanding something.

Comment (kayrus) [reply openstack/lb_v2_octavia.go:20]: I mean this: https://danp.net/posts/reducing-go-nesting/

**Review by kayrus — APPROVED** [review]
LGTM, thanks!

**Pull request merged**